docs/design-notes/qualifying-replicas-for-reads.md

```diff
@@ -0,0 +1,83 @@
+# Qualifying replicas for read operations
+
+## Intro
```

Comment:
Is this document supposed to cover all the aspects of choosing replicas?
If so it should also mention heat-weighted-load-balancing, snitches (not just "dynamicsnitch"), etc.
Or, if it's supposed to be a more specific document, the intro should explain it only covers part of the choosing-replicas-for-read story and that other parts are documented elsewhere.

Reply:
By the way, can you move this patch to be first instead of last? It can help reviewers understand the theory behind the code they'll see.

Reply:
Sure, I'll move it as the first patch. And I mean it to eventually cover all aspects, but it's out of scope for this series - I just hope that one day it will be filled with more information about other mechanisms that Scylla has.